controllers/argocd/deployment.go

```diff
@@ -34,6 +34,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// getArgoCDRepoServerReplicas will return the size value for the argocd-repo-server replica count.
```

Comment:
Since this return value is fed into `deploy.Spec.Replicas` which is a not pointer and has a default value of 1, we should not have our own default.  In other words, we should just pass the cr.Spec.Repo.Replicas value to `deploy.Spec.Replicas`.  If it is nil, pass nil to `deploy.Spec.Replicas`.

Reply:
// getArgoCDRepoServerReplicas will return the size value for the argocd-repo-server replica count if it has been set in argocd CR.  Otherwise, nil is returned.